python/ray/dashboard/datacenter.py

```diff
@@ -81,7 +90,7 @@ async def organize(cls):
         DataSource.core_worker_stats.reset(core_worker_stats)
```

Comment:
Is DataSource thread safe given now it's accessed in another thread as well

Reply:
not sure...

Reply:
+1 

These custom data-structures aren't thread-safe. If we're planning to run this on TPE we need to cover these with locks